Pass user-defined-trusted-dirs to glibc

We probably may need/use this for the CUDA support (see https://github.com/EESSI/compatibility-layer/issues/71), and possibly other use cases.

This can be done by modifying the glibc ebuild, but this is a bit cumbersome, since it has to be redone for every new version. We could, instead, use the following approach, which allows us to override variables per package (or even per version of a package):
https://groups.google.com/g/linux.gentoo.user/c/z1ZXuJkd0ro/m/9PUWFv8BXnUJ

I quickly tested it by adding a `$EPREFIX/etc/portage/bashrc`:
```
for MY_ENV in ${PN} ${P} ${PF}; do 
	if [ -f ${EPREFIX}/etc/portage/env.d/${CATEGORY}/${MY_ENV} ]; then 
		. ${EPREFIX}/etc/portage/env.d/${CATEGORY}/${MY_ENV} 
		fi 
	done

```
and a file `$EPREFIX/etc/portage/env.d/sys-libc/glibc`:
```
EXTRA_ECONF="--testme"
```

Running `emerge --ask --verbose sys-libs/glibc` then failed, as expected:
```
configure: error: unrecognized option: `--testme'
```

This can be easily included in the `etc/portage` dir of this repo, as our compatibility layer playbook already makes symlinks in `${EPREFIX}/etc/portage` to all files/dirs in this directory.

--- a/etc/portage/env/glibc-user-defined-trusted-dirs.conf
+++ b/etc/portage/env/glibc-user-defined-trusted-dirs.conf
@@ -1,0 +1,1 @@
+EXTRA_EMAKE="user-defined-trusted-dirs=/opt/eessi/lib64"

--- a/etc/portage/env/glibc-user-defined-trusted-dirs.conf
+++ b/etc/portage/env/glibc-user-defined-trusted-dirs.conf
@@ -1,1 +1,1 @@
-EXTRA_EMAKE="user-defined-trusted-dirs=/opt/eessi/lib64"
+EXTRA_EMAKE="user-defined-trusted-dirs=/opt/eessi/lib"

--- a/etc/portage/package.env
+++ b/etc/portage/package.env
@@ -1,0 +1,1 @@
+sys-libs/glibc glibc-user-defined-trusted-dirs.conf

